src/rebar_prv_clean.erl

```diff
@@ -46,12 +47,12 @@ do(State) ->
     %% Need to allow global config vars used on deps
     %% Right now no way to differeniate and just give deps a new state
     EmptyState = rebar_state:new(),
-    clean_apps(EmptyState, DepApps),
+    clean_apps(EmptyState, Providers, DepApps),
 
     Cwd = rebar_dir:get_cwd(),
-    rebar_hooks:run_compile_hooks(Cwd, pre_hooks, clean, State),
-    clean_apps(State, ProjectApps),
-    rebar_hooks:run_compile_hooks(Cwd, post_hooks, clean, State),
+    rebar_hooks:run_all_hooks(Cwd, pre, ?PROVIDER, Providers, State),
+    clean_apps(State, Providers, ProjectApps),
+    rebar_hooks:run_all_hooks(Cwd, post, ?PROVIDER, Providers, State),
```

Comment:
Aren't these made redundant -- or double through the ones in `clean_apps`, or does this one apply to the top level, then the other to apps individually?